### PR TITLE
GH-72 - DA Config

### PR DESCRIPTION
**Pull request by auniverseaway**
* Ability to add arbitrary sheet-based configs to **org** and **repo** folders via a new `/config` view.
* Refactored `getPathUtils.js` to better clarify `/my-item/` (folder) vs `/my-item` (html) vs `/my-item.json` (json)
* Added tests for `getPathUtils.js` and `constants.js`.
* Made `constants.js` more testable as it was caching a property as a result of ESM side-effects
* Minor refactoring of `da-browse` to support config icon in breadcrumb.
* Minor refactoring of `da-title` to support **save** vs **preview & publish**.

Resolves: GH-72

Screenshot 2024-03-03 at 2 07 06 PM

Screenshot 2024-03-03 at 2 08 23 PM

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

## Release notes
* Adds ability to store structured config data at the organization and project level.
* Improved sheet reliability

### Test URLs
https://config--da-live--adobe.hlx.page/config#/adobecom/da-milo-college/

### Screenshots
Screenshot 2024-03-11 at 11 53 55 AM

Screenshot 2024-03-11 at 11 54 18 AM

### CODE REVIEW

File: test/blocks/shared/constants.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { expect } from '@esm-bundle/chai';
+import { getDaAdmin, COLLAB_ORIGIN } from '../../../blocks/shared/constants.js';
+
+import '../../milo.js';
+
+describe('DA Admin', () => {
+  describe('Set DA admin default', () => {
```
Comment (bosschaert) [inline test/blocks/shared/constants.test.js:7]: More a question than a comment... I thought that the actual Mocha tests were normally put in `it(..)` calls while the test suite sections/groupings are in `describe(...)` calls. I see that you have the actual tests also in `describe` calls. Any particular reason for this?

Comment (auniverseaway) [reply test/blocks/shared/constants.test.js:7]: You're right. Bad copy / paste. It should be `it`. Will fix. Thanks @bosschaert!

Comment (auniverseaway) [reply test/blocks/shared/constants.test.js:7]: @bosschaert this should be fixed now. Thanks for reviewing.

**Review by bosschaert — APPROVED** [review]
The tests look good 😄

**Comment by auniverseaway** [comment]
@karlpauls @bosschaert this is ready to review again. With the squashing of commits for collab awareness, your reviews got dismissed.

@karlpauls the pre-existing bug you found is still present (in main and in this branch).

**Review by karlpauls — DISMISSED** [review]

**Review by bosschaert — APPROVED** [review]

**Review by karlpauls — APPROVED** [review]

**Review by bosschaert — APPROVED** [review]

**Pull request merged**